.github/workflows/run-playwright-on-pr.yaml

```diff
@@ -0,0 +1,71 @@
+name: Playwright tests on pr
+on:
+  pull_request:
+    branches: [master]
+    types: [opened, synchronize, reopened]
+    paths:
+      - 'frontend/**'
+      - '!frontend/stats/**'
+      - 'backend/**'
+      - '.github/workflows/run-playwright-on-pr.yaml'
+      - 'docker-compose.yml'
+      - 'Dockerfile'
+      - 'gitea/**'
+      - 'package.json'
+      - 'development/**'
+  workflow_dispatch:
+
+jobs:
+  playwright-tests:
+    name: Build environment and run e2e test
+    timeout-minutes: 25
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Generate .env file
+        run: |
+          echo PLAYWRIGHT_TEST_APP=autodeploy-v3 >> .env
+          echo DEVELOP_APP_DEVELOPMENT=0 >> .env
+          echo DEVELOP_RESOURCE_ADMIN=0 >> .env
+          echo DEVELOP_BACKEND=0 >> .env
+          echo DEVELOP_DASHBOARD=0 >> .env
+          echo DEVELOP_PREVIEW=0 >> .env
+          echo GITEA_ADMIN_PASS=g9wDIG@6gf >> .env
+          echo GITEA_ADMIN_USER=localg1iteaadmin >> .env
+          echo GITEA_CYPRESS_USER=cypress_testuser >> .env
+          echo GITEA_CYPRESS_PASS=g9wDIG@6gf >> .env
+          echo GITEA_ORG_USER=ttd >> .env
+          echo POSTGRES_PASSWORD=kyeDIG@eip >> .env
+          echo COMMIT= >> .env
+          echo IGNORE_DOCKER_DNS_LOOKUP=true >> .env
+
+      - name: Build all images
+        run: |
+          docker compose build --no-cache
+
+      - name: Install node
+        uses: actions/setup-node@v4
+        with:
+          cache: 'yarn'
+
+      - name: Run setup.js script
+        run: |
+          node ./development/setup.js
```

Comment:
Can we consider reusing the same artifact/build that already exists? In the current state, both Cypress and Playwright require separate builds. It would be sufficient to build the application once and then reuse the same artifact across both.

Should we look into the GitHub Docs and see if it's possible to share artifact between workflows? 😊